pkg/pool-manager/pool_test.go

```diff
@@ -414,4 +416,61 @@ var _ = Describe("Pool", func() {
 		})
 	})
 
+	Describe("Multus Network Annotations API Tests", func() {
+		pod := v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "testPod", Namespace: "default"}}
+		poolManager := &PoolManager{}
+
+		Context("when pool-manager is configured with available addresses", func() {
+			Context("Pod is created with networks annotation with mac-address", func() {
+				BeforeEach(func() {
+					poolManager = createPoolManager("02:00:00:00:00:00", "02:00:00:00:00:02", &vmConfigMap)
+					Expect(poolManager).ToNot(Equal(nil), "should create pool-manager")
+				})
+
+				table.DescribeTable("should allocate mac-address correspond to the one specified in the networks annotation",
+					func(networkRequestAnnotation string) {
+						pod.Annotations = map[string]string{networksAnnotation: fmt.Sprintf("%s", networkRequestAnnotation)}
+
+						By("Request specific mac-address by adding the address to the networks pod annotation")
+						err := poolManager.AllocatePodMac(&pod)
+						Expect(err).ToNot(HaveOccurred(), "should allocate mac address and ip address correspond to networks annotation")
+
+						By("Convert obtained networks annotation JSON to multus.NetworkSelectionElement array")
+						obtainedNetworksAnnotationJson := pod.Annotations[networksAnnotation]
+						obtainedNetworksAnnotation := []multus.NetworkSelectionElement{}
+						err = json.Unmarshal([]byte(obtainedNetworksAnnotationJson), &obtainedNetworksAnnotation)
+						Expect(err).ToNot(HaveOccurred(), "should convert obtained annotation as json to multus.NetworkSelectionElement")
+
+						By("Convert expected networks annotation JSON to multus.NetworkSelectionElement array")
+						expectedNetworksAnnotation := []multus.NetworkSelectionElement{}
+						err = json.Unmarshal([]byte(networkRequestAnnotation), &expectedNetworksAnnotation)
+						Expect(err).ToNot(HaveOccurred(), "should convert expected annotation as json to multus.NetworkSelectionElement")
+
+						By("Compare between each obtained and expected network request according to the mac-address")
+						for _, obtainedNetwork := range obtainedNetworksAnnotation {
+							for _, expectedNetwork := range expectedNetworksAnnotation {
```

Comment:
Either check size before loop or convert slices into []string with the macRequest so you can use ContaineElements and it will already check the size for you.

Reply:
I will add a break for the end of the if that should prevent overlapping, once we bump Gomega  we can use ContaineElements